source/extensions/filters/http/local_ratelimit/local_ratelimit.cc

```diff
@@ -73,20 +71,37 @@ FilterConfig::FilterConfig(
     throw EnvoyException("local rate limit token bucket must be set for per filter configs");
   }
 
+  absl::Status creation_status;
+  rate_limit_config_ = std::make_unique<Filters::Common::RateLimit::RateLimitConfig>(
+      config.rate_limits(), context, creation_status);
+  THROW_IF_NOT_OK_REF(creation_status);
+
+  if (rate_limit_config_->empty()) {
+    if (!config.descriptors().empty()) {
+      ENVOY_LOG_FIRST_N(
+          warn, 20,
+          "'descriptors' is set but no valid 'rate_limits' in LocalRateLimit config. Note that "
+          "'descriptors' only makes sense when 'rate_limits' in LocalRateLimit config or route "
+          "config is specified. And please take the 'rate_limits' in the LocalRateLimit config as "
+          "priority because the 'rate_limits' in the route config may be ignored by the filter in "
+          "the future.");
```

Comment:
I think something like this makes more sense?

```suggestion
          "'descriptors' is set but only used by route configuration. Please configure the local rate limit filter using the embedded 'rate_limits' field as route level configuration for local rate limits will be ignored in the future.");
```